### DISCUSSION TITLE
Copying hyphens before line breaks?

### DISCUSSION THREAD

**Discussion by vipatel10** [sumatrapdfreader/sumatrapdf#2705] [Q&A]
It seems that when I copy from a PDF in Sumatra a paragraph  with justified formatting that splits words at the end of the line with a hyphen, the hyphen is not copied.  For example, if this is the paragraph in the PDF:

The ablest administrator of his day and a devout
Lutheran, he intervened to support the empire’s Prot­-
estants.

When it's copied, only this is sent to the clipboard:

The ablest administrator of his day and a devout
Lutheran, he intervened to support the empire’s Prot­
estants.  

The hyphen is missing.  Is this intentional?  I'm wondering if there's a way to have the hyphen copied because when I'm copying from a PDF to a word doc to make notes, I'd rather have the hyphen there so I can run a tool to remove the line break and the hyphen to join the two halves of the word together.  In the example above, to join "Prot-" and "estants" into Protestants.

**Comment by vipatel10** [comment]
Update:  It seems this only happens in certain parts of the PDF.  In other parts of the PDF, the hyphen copies as I was hoping it would above.  Now I'm totally confused why this would be.

**Reply by GitHubRulesOK** [reply]
Hyphens can be soft whilst others hard, and thus guess that's the difference

[A soft hypen may ] becomes visible only after word wrapping at the end of a line.

However there is no word-wrap in a pdf (each line part is independent of all other parts, thus 1 character can be a word or a paragraph)

Hyphens can be problematic when not displayed as they need to be used for bookmarks or for search to work see an example at end of https://github.com/sumatrapdfreader/sumatrapdf/issues/2647#issuecomment-1144001451

**Comment by kjk** [comment]
Text in pdf is "character X at position Y". It's not really a word or a sentence the way you might think of it.

Therefore copying & pasting text from PDF into a a sentence is a bunch of heuristics. "Is this letter close to that letter? If yes, they're probably part of the same word" etc.

Therefore it all depends on the specific PDF and without access to that specific PDF I can't say anything beyond the generic description above.
